ignite/metrics/metrics_lambda.py

```diff
@@ -64,17 +66,47 @@ def compute(self) -> Any:
         return self.function(*materialized, **materialized_kwargs)
 
     def _internal_attach(self, engine: Engine) -> None:
+        self.engine = weakref.ref(engine)
         for index, metric in enumerate(itertools.chain(self.args, self.kwargs.values())):
             if isinstance(metric, MetricsLambda):
                 metric._internal_attach(engine)
             elif isinstance(metric, Metric):
+                # NB : metrics is attached partially
+                # We must not use is_attached() but rather if these events exist
                 if not engine.has_event_handler(metric.started, Events.EPOCH_STARTED):
                     engine.add_event_handler(Events.EPOCH_STARTED, metric.started)
                 if not engine.has_event_handler(metric.iteration_completed, Events.ITERATION_COMPLETED):
                     engine.add_event_handler(Events.ITERATION_COMPLETED, metric.iteration_completed)
 
     def attach(self, engine: Engine, name: str) -> None:
-        # recursively attach all its dependencies
+        # recursively attach all its dependencies (partially)
         self._internal_attach(engine)
         # attach only handler on EPOCH_COMPLETED
         engine.add_event_handler(Events.EPOCH_COMPLETED, self.completed, name)
+
+    def detach(self, engine: Engine) -> None:
+        # remove from engine
+        super().detach(engine)
+        # invalidate weakref
+        self.engine = None
+
+    def is_attached(self, engine: Engine) -> bool:
+        # check recursively the dependencies
+        return super().is_attached(engine) and self._internal_is_attached(weakref.ref(engine))
```

Comment:
```suggestion
        return super(MetricsLambda, self).is_attached(engine) and self._internal_is_attached(weakref.ref(engine))
```

Reply:
Can we simply put engine here instead of weakref ?
```
self._internal_is_attached(weakref.ref(engine))
```

Reply:
Sure we can